code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -379,7 +379,27 @@
 	var/static/list/color_list = list("#FF0000","#FF8C00","#FFFF00","#00FF00","#00BFFF","#0000FF","#9400D3")
 	icon_state = "rainbow_laser"
 	item_state = null
+	slot_flags = SLOT_BELT
+	charge_cost = 100
+	cell_type = "/obj/item/weapon/cell"
 	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/guns.dmi', "right_hand" = 'icons/mob/in-hand/right/guns.dmi')
+	var/pumping = 0
+
+/obj/item/weapon/gun/energy/laser/rainbow/attack_self(mob/user as mob)
+
+	if(pumping || !power_supply)
+		return
+	pumping = 1
+	power_supply.charge = min(power_supply.charge + 200,power_supply.maxcharge)
```

Comment:
DEFINES for amount added

Reply:
also you could move this into the else part since you don't need to do this calculation if it's at max

Reply:
yea but then you can't keep honking the airhorn